### PR TITLE
feat(*): add repo commands

### PR DESCRIPTION
**Pull request by technosophos**
This adds the following commands:

```
helm repo add [lname] [gitURL]
helm repo rm [lname]
helm repo list
```

This addresses #118 #111 and #109.

### CODE REVIEW

File: action/search.go [hunk 1]
```diff
@@ -13,52 +13,70 @@ import (
...
+	}
 
 	for _, dir := range dirs {
-		chart, err := chart.LoadChartfile(filepath.Join(dir, "Chart.yaml"))
+		cname := filepath.Join(table, filepath.Base(dir))
+		chrt, err := chart.LoadChartfile(filepath.Join(dir, "Chart.yaml"))
 
 		if err != nil {
-			log.Warn("failed to load Chart.yaml: %v", err)
+			//log.Warn("Failed to load Chart.yaml: %v", err)
```
Comment (mboersma) [inline action/search.go]: It's ok not to log the error here?

File: Makefile [hunk 2]
```diff
@@ -47,10 +47,10 @@ else
 endif
 
 quicktest:
-	go test ./. ./manifest ./action ./log ./chart ./dependency
+	go test ./. ./manifest ./action ./log ./chart./dependency ./repo
```
Comment (adamreese) [inline Makefile]: need a space `./chart ./dependency`

File: action/search.go (hunk 1 above)
Comment (technosophos) [inline action/search.go]: Right. It is not an error for a directory to not be a `chart`. This was generating a lot of spurious errors for directories like `.git`. (I actually removed the log message in one of the other PRs somewhere in here... so even the comment will go away)

**Comment by adamreese** [comment]
@technosophos what is the thought behind a `Repofile`?  A generic helm config file would scale better for new features that require persistent settings

**Comment by technosophos** [comment]
Fixed `quicktest` and replaced the commented log message with a comment explaining why we intentionally skip.

**Comment by mboersma** [comment]
Groovy:
```console
$ helm up
---> Creating /Users/matt/.helm/cache/Repos.yaml
---> Updating charts
---> Done
$ helm repo add bacongobbler https://github.com/bacongobbler/charts
---> Cloning into '/Users/matt/.helm/cache/bacongobbler'...
$ helm repo list
	charts*	https://github.com/deis/charts
	bacongobbler	https://github.com/bacongobbler/charts
$ helm fetch bacongobbler/redis-standalone
---> Fetched chart into workspace /Users/matt/.helm/workspace/charts/redis-standalone
---> Done
```

**Comment by technosophos** [comment]
@adamreese what's the use case for a general settings file? I could probably expand the Repofile to that, but I don't know what else I'd put in there.

I wasn't really intending for a user to edit the Repofile by hand unless to recover from something catastrophic, so I wasn't really intending it to be used like a general settings file. But I guess I could see it as sorta playing the same role as a `.git/config` file. Is that along the lines of what you were thinking?

**Comment by adamreese** [comment]
Yes, It could be more maintainable on the dev side when feature x, y, and z are out and all need config files.

Just my 2 cents...

**Comment by technosophos** [comment]
So what if we did it like this:

Inside of `$HELM_HOME`, add a `config.yaml` file. The structure would look like this:

```
repos:
  default: charts
  tables:
    - name: charts
      repo: https://github.com/deis/charts
```

That would leave the top-level namespace open for other settings alongside `repos`.

Would that be good, @adamreese ?

**Comment by adamreese** [comment]
@technosophos superb

**Comment by technosophos** [comment]
@adamreese Great idea. I'll revise this branch to do that. No point in merging this then having to break things again.

**Comment by technosophos** [comment]
Closing this and pushing in a new version on a separate PR

**Pull request closed without merging**